synapse/handlers/room_member.py

```diff
@@ -1190,6 +1190,13 @@ async def update_membership_locked(
             origin_server_ts=origin_server_ts,
         )
 
+    async def check_user_membership(self, user_id: str, room_id: str) -> None:
+        result: Optional[Tuple[Optional[str], Optional[str]]] = await self.store.get_local_current_membership_for_user_in_room(user_id=user_id, room_id=room_id)
```

Comment:
No need to add a type here. `mypy` can already assume from what `get_local_current_membership_for_user_in_room(...)` returns.

```suggestion
        result = await self.store.get_local_current_membership_for_user_in_room(user_id=user_id, room_id=room_id)
```